src/hotspot/share/opto/escape.cpp

```diff
@@ -694,72 +694,66 @@ void ConnectionGraph::add_final_edges(Node *n) {
     case Op_LoadPLocked: {
       // Using isa_ptr() instead of isa_oopptr() for LoadP and Phi because
       // ThreadLocal has RawPtr type.
-      const Type* t = _igvn->type(n);
-      if (t->make_ptr() != NULL) {
-        Node* adr = n->in(MemNode::Address);
-        add_local_var_and_edge(n, PointsToNode::NoEscape, adr, NULL);
-        break;
-      }
-      ELSE_FAIL("Op_LoadP");
+      assert(_igvn->type(n)->make_ptr() != NULL,
+             "constrain must match when firstly add to cg");
```

Comment:
I would change the assert message to `Unexpected node type`. Same for other asserts below.